bjoern/request.c

```diff
@@ -121,50 +121,70 @@ on_path(http_parser* parser, const char* path, size_t len)
 {
   if(!(len = unquote_url_inplace((char*)path, len)))
     return 1;
-  _set_header_free_value(_PATH_INFO, _Unicode_FromStringAndSize(path, len));
+  _set_or_append_header(_PATH_INFO, path, len);
   return 0;
 }
 
 static int
 on_query_string(http_parser* parser, const char* query, size_t len)
 {
-  _set_header_free_value(_QUERY_STRING, _Unicode_FromStringAndSize(query, len));
+  _set_or_append_header(_QUERY_STRING, query, len);
   return 0;
 }
 
 static int
 on_header_field(http_parser* parser, const char* field, size_t len)
 {
-  if(PARSER->value.data) {
-    /* Store previous header and start a new one */
-    _set_header_from_http_header();
-  } else if(PARSER->field.data) {
-    UPDATE_LENGTH(field);
+  if(PARSER->last_was_data) {
```

Comment:
Would you mind adding a comment here what this case means? Along the lines of "we saw a new header field but still have the old value buffered; put them into the environ dict".

Reply:
some comments added around this and others